src/TaskGroup.h

```diff
@@ -125,6 +128,8 @@ class TaskGroup : public HasTaskSet {
   std::set<TaskGroup*> children_;
 
   uint32_t serial;
+
+  ThreadDb thread_db_;
```

Comment:
I think we should make this a `unique_ptr` so that we don't need to include `ThreadDB.h` here and `ThreadDB` will not need to be instantiated during recording, which doesn't need this.